### ISSUE TITLE
How to resolve webview build failed?

### ISSUE THREAD

**Issue by yangsh91** [flutter/flutter#88419]
I tried flutter webview watching on youtube ( https://www.youtube.com/watch?v=5u3pZZ6uvDo&t=303s )
However, I don't know. Why are occur build failed. 

I try entered some command .. flutter clean .. flutter pub get..

```
yangsanghun@Yangui-MacBookAir webviewexample % flutter run
Launching lib/main.dart on iPhone 12 Pro in debug mode...
Running pod install...                                              3.5s
Running Xcode build...                                                  
Xcode build done.                                            8.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    duplicate symbol '_OBJC_CLASS_$_FLTWebViewFlutterPlugin' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWebViewFlutterPlugin.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWebViewFlutterPlugin 2.o
    duplicate symbol '_OBJC_METACLASS_$_FLTWebViewFlutterPlugin' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWebViewFlutterPlugin.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWebViewFlutterPlugin 2.o
    duplicate symbol '_OBJC_IVAR_$_FLTWKProgressionDelegate._methodChannel' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate 2.o
    duplicate symbol '_FLTWKEstimatedProgressKeyPath' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate 2.o
    duplicate symbol '_OBJC_CLASS_$_FLTWKProgressionDelegate' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate 2.o
    duplicate symbol '_OBJC_METACLASS_$_FLTWKProgressionDelegate' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/FLTWKProgressionDelegate 2.o
    duplicate symbol '_OBJC_CLASS_$_FLTJavaScriptChannel' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler 2.o
    duplicate symbol '_OBJC_METACLASS_$_FLTJavaScriptChannel' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler 2.o
    duplicate symbol '_OBJC_IVAR_$_FLTJavaScriptChannel._methodChannel' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler 2.o
    duplicate symbol '_OBJC_IVAR_$_FLTJavaScriptChannel._javaScriptChannelName' in:
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler.o
        /Users/yangsanghun/Library/Developer/Xcode/DerivedData/Runner-alabyyhhwxgaxxbjccvtytwljtgy/Build/Intermediates.noindex/Pods.build/Debug
        -iphonesimulator/webview_flutter.build/Objects-normal/x86_64/JavaScriptChannelHandler 2.o
    ld: 10 duplicate symbols for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    warning: Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve
    this, select a development team in the Runner editor. (in target 'Runner' from project 'Runner')

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro.
```

**Comment by darshankawar** [comment]
@yangsh91 
It's failing due to `duplicate symbol '_OBJC_CLASS_$_FLTWebViewFlutterPlugin`. 
Do you somehow have iCloud running in the back ? that maybe causing the generating files to duplicate.

**Comment by yangsh91** [comment]
> @yangsh91
> It's failing due to `duplicate symbol '_OBJC_CLASS_$_FLTWebViewFlutterPlugin`.
> Do you somehow have iCloud running in the back ? that maybe causing the generating files to duplicate.

I don't know, How to check iCloud running in the back?

**Comment by darshankawar** [comment]
@yangsh91 
Can you provide `flutter doctor -v`, `pubspec.yaml` ?

**Comment by yangsh91** [comment]
> @yangsh91
> Can you provide `flutter doctor -v`, `pubspec.yaml` ?


flutter doctor -v

flutter doctor -v
[✓] Flutter (Channel master, 2.5.0-7.0.pre.130, on macOS 11.2.3 20D91 darwin-x64, locale ko)
    • Flutter version 2.5.0-7.0.pre.130 at /Users/yangsanghun/Documents/developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d83fba1a22 (5시간 전), 2021-08-18 03:02:04 -0400
    • Engine revision a27da3eeb6
    • Dart version 2.15.0 (build 2.15.0-24.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/yangsanghun/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • iPhone 12 Pro (mobile) • 22F845FC-3442-4FBA-B4D8-56877768E2D0 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 92.0.4515.159



pubspec.yaml is
name: webviewexample
description: A new Flutter project.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2

dev_dependencies:
  flutter_test:
    sdk: flutter

  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^1.0.0
  webview_flutter: ^2.0.12

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

**Comment by darshankawar** [comment]
@yangsh91 
Also provide below files:

```
pubspec.lock
ios/Podfile
ios/Podfile.lock
```

**Comment by danagbemava-nc** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
